packages/atlas/src/hooks/useReactionTransactions.ts

```diff
@@ -0,0 +1,157 @@
+import { ConsoleLogger } from '@/utils/logs'
+
+export const useReactionTransactions = () => {
+  const { activeMemberId } = useUser()
+  const { joystream, proxyCallback } = useJoystream()
+  const [commentInputProcessing, setCommentInputProcessing] = useState(false)
+  const [videoReactionProcessing, setVideoReactionProcessing] = useState(false)
+  const [highlightedComment, setHighlightedComment] = useState<string | null>(null)
+  const handleTransaction = useTransaction()
+  const [processingCommentReactionId, setProcessingCommentReactionId] = useState<string | null>(null)
+
+  const client = useApolloClient()
+
+  const refetchComments = useCallback(
+    () =>
+      client.refetchQueries({
+        include: [GetCommentsDocument],
+      }),
+    [client]
+  )
+
+  useEffect(() => {
+    if (!highlightedComment) {
+      return
+    }
+    const timeout = setTimeout(() => {
+      setHighlightedComment(null)
+    }, 3000)
+
+    return () => clearTimeout(timeout)
+  })
+
+  const reactToComment = useCallback(
+    async (commentId: string, reactionId: ReactionId) => {
+      if (!joystream || !activeMemberId) {
+        ConsoleLogger.error('No joystream instance')
+        return
+      }
+
+      setProcessingCommentReactionId(commentId + `-` + reactionId.toString())
+
+      return handleTransaction({
+        txFactory: async (updateStatus) =>
+          (await joystream.extrinsics).reactToVideoComment(
+            activeMemberId,
+            commentId,
+            reactionId,
+            proxyCallback(updateStatus)
+          ),
+        minimized: {
+          signErrorMessage: 'Failed to react to comment',
+        },
+        onTxSync: async () => {
+          await refetchComments()
+          setProcessingCommentReactionId(null)
+        },
+        onError: () => {
+          setProcessingCommentReactionId(null)
+        },
+      })
+    },
+    [activeMemberId, handleTransaction, joystream, proxyCallback, refetchComments]
+  )
+
+  const addComment = useCallback(
+    async (videoId: string, commentBody: string, parentCommentId?: string) => {
+      if (!joystream || !activeMemberId) {
+        ConsoleLogger.error('no joystream or active member')
+        return
+      }
+
+      return handleTransaction({
+        preProcess: () => {
+          setCommentInputProcessing(true)
+        },
+        txFactory: async (updateStatus) =>
+          (await joystream.extrinsics).createVideoComment(
+            activeMemberId,
+            videoId,
+            commentBody,
+            parentCommentId || null,
+            proxyCallback(updateStatus)
+          ),
+        onTxSync: async ({ block }) => {
+          const refetchResult = await refetchComments()
+          const newCommentsQueryResult = refetchResult[0] as GetCommentsQueryHookResult
+          const newCommentId = newCommentsQueryResult.data?.comments.find(
+            (comment) => comment.commentcreatedeventcomment?.[0].inBlock === block
+          )?.id
```

Comment:
What if there's multiple comments added in the same block? I would suggest to skip this part in this PR as this will be very tricky to get right. Once we have https://github.com/Joystream/atlas/issues/2629, this should be much easier

Reply:
Or I guess we can leave it but add a TODO that this needs to be changed